SIPS/sip-26.md

```diff
@@ -0,0 +1,168 @@
+The Account Router exposes the [`snap_manageAccounts`][snap-manage-accs] method
+to allow account Snaps to register, remove, and update accounts.
+
+For example, this code can be used by an Account Snap to register a new account
+with the Account Router:
+
+```typescript
+// This will notify the Account Router that a new account was created, and the
+// Account Router will register this account as available for signing requests
+// using the `eth_signTypedData_v4` method.
+await snap.request({
+  method: 'snap_manageAccounts',
+  params: {
+    method: 'notify:accountCreated',
+    params: {
+      account: {
+          id: '74bb3393-f267-48ee-855a-2ba575291ab0',
+          type: 'eip155:eoa',
+          address: '0x1234567890123456789012345678901234567890',
+          methods: ['eth_signTypedData_v4'],
+          options: {},
+      },
+    },
+  },
+});
+```
+
+Similar events are available to notify about the removal and update of
+accounts: `notify:accountRemoved` and `notify:accountUpdated`.
+
+Additionally, the Account Router expects the Account Snap to implement the
+Keyring API so it can forward signing requests to it through the
+[`keyring_submitRequest`][submit-request] method.
+
+### Account Address Resolution Snaps
+
+The Account Address Resolution Snaps are responsible for extracting the account
+address that should receive the signing request from the request object. This
+is accomplished by exposing the `resolveAccountAddress` method to the Account
+Router.
+
+```typescript
+/**
+ * Returns the address of the account that should handle the signing request.
+ *
+ * @param request - The request object.
+ * @returns The account address or `undefined` if the address could not be
+ * resolved.
+ */
+function resolveAccountAddress(request: MultichainRequest): string | undefined;
+```
+
+There must be only one Account Resolution Snap registered per chain to prevent
+ambiguity in the account resolution process.
+
+To identify which Account Resolution Snap should be used for a given request,
+Account Address Resolution Snaps MUST list the supported chains in their
+manifest file using the [CAIP-2][caip-2] format for chain IDs:
+
+```json5
+"initialPermissions": {
+  "endowment:account-address-resolver": {
+    "chains": [
+      "<chain_id_1>",
+      "<chain_id_2>",
+      // ...
+    ]
+  }
+}
+```
+
+Note that the `reference` part of the [CAIP-2][caip-2] chain ID can be a `*`
+wildcard to match any chain ID of a given namespace (e.g., `eip155:*`).
+
+The RPC Router subscribes to Snap installation events to detect new Account
+Address Resolution Snaps and updates its internal routing table accordingly.
+
+### Protocol Snaps
+
+Protocol Snaps implement and expose methods that do not require an account to
+execute and MUST list their supported methods in their manifest file:
+
+```json5
+"initialPermissions": {
+  "endowment:protocol-methods": {
+    "chains": {
+      "<chain_id_1>": [
+        // List of supported methods
+      ]
+    }
```

Comment:
this should probably resemble caip-25 and the permission we just created and migrated the old account permissions to:

```

   "eip155:1": {
      "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
      "notifications": []
   },
```